File: Src/Services/Identity/LookupNormalizer.cs
using Microsoft.AspNetCore.Identity;

namespace MusicPlayerBackend.Services.Identity;

public sealed class LookupNormalizer : ILookupNormalizer
{
    public string? NormalizeName(string? name)
    {
        return name?.ToUpperInvariant();
    }

    public string? NormalizeEmail(string? email)
    {
        return email?.ToUpperInvariant();
    }
}

File: Src/Identity/LookupNormalizer.fs
namespace MusicPlayerBackend.Identity

open Microsoft.AspNetCore.Identity
open MusicPlayerBackend.Common

[<Sealed>]
type LookupNormalizer() =
    interface ILookupNormalizer with
        member this.NormalizeEmail(email) =
            email
            |> Option.ofStringW
            |> StringOption.toUpperInv
            |> Option.toObj

        member this.NormalizeName(name) =
            name
            |> Option.ofStringW
            |> StringOption.toUpperInv
            |> Option.toObj
